src/syntax/sedlex.ml

```diff
@@ -117,6 +129,89 @@ let transition (state : state) =
   Array.sort (fun (c1, _) (c2, _) -> compare c1 c2) t;
   t
 
+(* Restore NFA path from DFA path *)
+
+type trans_case = {
+  curr_state : int;
+  curr_node : int;
+  prev_state : int;
+  prev_node : int;
+  char_set : Sedlex_cset.t;
+  actions : action list;
+}
+
+type final_case = { curr_node : int; actions : action list }
+
+let compile_traces states (start, final) =
+  let counter = ref 0 in
+  let nodes_idx = Hashtbl.create 31 in
+  let rec aux node =
+    if not (Hashtbl.mem nodes_idx node) then begin
+      let i = !counter in
+      incr counter;
+      Hashtbl.add nodes_idx node i;
+      List.iter aux node.eps;
+      List.iter (fun (_, next) -> aux next) node.trans
+    end
+  in
+  aux start;
+  let append_action actions = function
+    | None -> actions
+    | Some action -> action :: actions
+  in
+  let first_node = Hashtbl.find nodes_idx final in
+  let trans_cases =
+    let cases = Hashtbl.create 31 in
+    Hashtbl.iter
+      (fun from_state j ->
+        Hashtbl.iter
+          (fun to_state i ->
+            List.iter
+              (fun from_node ->
+                try
```

Comment:
is this try-with necessary ?

Reply:
It caused some problem when I removed `nodes_idx`. Now it's fixed.

Reply:
Can you elaborate on the issue and explain what are the irrelevant nodes ?